src/new-client/src/App/Trades/TradesGrid/TradesGrid.tsx

```diff
@@ -95,56 +119,83 @@ export const TradesGrid: React.FC = () => {
     }
   }
 
+  return (
+    <TableBodyRow
+      key={trade.tradeId}
+      highlight={trade.tradeId === highlightedRow}
+      onClick={() => tryBroadcastContext(trade.symbol)}
+    >
+      <StatusIndicator status={trade.status} aria-label={trade.status} />
+      <Spacer aria-hidden={true} />
+      {colFields.map((field, i) => (
+        <TableBodyCell
+          key={field}
+          numeric={
+            colConfigs[field].filterType === "number" && field !== "tradeId"
+          }
+          rejected={trade.status === "Rejected"}
+          relativeWidth={colConfigs[field].width}
+        >
+          {colConfigs[field].valueFormatter?.(trade[field]) ?? trade[field]}
+        </TableBodyCell>
+      ))}
+    </TableBodyRow>
+  )
+}
+
+export const TradesGrid: React.FC<any> = ({ height = 200, width = 200 }) => {
+  const trades = useTableTrades()
+  const highlightedRow = useTradeRowHighlight()
+  const ref = useRef(null)
+
   return (
     <TableWrapper>
       <Table>
         <caption id="trades-table-heading" className="visually-hidden">
           Reactive Trader FX Trades Table
         </caption>
-        <TableHead>
-          <TableHeadRow>
-            <StatusIndicatorSpacer scope="col" aria-label="Trade Status" />
-            {colFields.map((field) => (
-              <TableHeadCellContainer key={field} field={field} />
-            ))}
-          </TableHeadRow>
-        </TableHead>
-        <tbody role="grid">
-          {trades.length ? (
-            trades.map((trade) => (
-              <TableBodyRow
-                key={trade.tradeId}
-                highlight={trade.tradeId === highlightedRow}
-                onClick={() => tryBroadcastContext(trade.symbol)}
-              >
-                <StatusIndicator
-                  status={trade.status}
-                  aria-label={trade.status}
+        <VirtuosoTable role="grid" width={width}>
+          <TableHead>
+            <TableHeadRow>
+              {/* <StatusIndicatorSpacer scope="col" aria-label="Trade Status" /> */}
+              <StatusIndicatorSpacer scope="col" aria-label="Trade Status" />
+              <SpacerHeader aria-hidden={true} />
+              {colFields.map((field) => (
+                <TableHeadCellContainer key={field} field={field} />
+              ))}
+            </TableHeadRow>
+          </TableHead>
+          {trades.length !== 0 && (
+            //Pending to handle the height in order to
+            //@ts-ignore
+            <Virtuoso
+              ref={ref}
```

Comment:
I'm not familiar with virtuoso, but these seems like it might be redundant?

Reply:
It is as for now, fixed.

Reply:
I'm confused - the ref is needed?

Reply:
Sorry, thought you meant the //@ts-ignore, removed the ref, wasn't necesary after all